src/Build/Evaluation/ProjectRootElementCache.cs

```diff
@@ -69,20 +69,21 @@ internal class ProjectRootElementCache : ProjectRootElementCacheBase
         /// If this number is increased much higher, the datastructure may
         /// need to be changed from a linked list, since it's currently O(n).
         /// </remarks>
-        private static readonly int s_maximumStrongCacheSize = 200;
+        private static readonly int s_maximumStrongCacheSize = !string.IsNullOrEmpty(Environment.GetEnvironmentVariable("MSBUILDPROJECTROOTELEMENTCACHESIZE")) ?
+            Convert.ToInt32(Environment.GetEnvironmentVariable("MSBUILDPROJECTROOTELEMENTCACHESIZE"), NumberFormatInfo.InvariantInfo)  : 200;
```

Comment:
I'm tempted to propose:
```suggestion
        private static readonly int s_maximumStrongCacheSize = !string.IsNullOrEmpty((string s = Environment.GetEnvironmentVariable("MSBUILDPROJECTROOTELEMENTCACHESIZE"))) ?
            Convert.ToInt32(s, NumberFormatInfo.InvariantInfo) : 200;
```

Honestly not sure which I like better. I like inlining most of these, but this one is messy.

Reply:
Actually, Environment.GetEnvironmentVariable returns null if it isn't defined; how about:
```suggestion
        private static readonly int s_maximumStrongCacheSize =
            Convert.ToInt32(Environment.GetEnvironmentVariable("MSBUILDPROJECTROOTELEMENTCACHESIZE") ?? "200", NumberFormatInfo.InvariantInfo);
```
?

Reply:
I also thought that, but what if the environment variable is an empty string value?

Reply:
On Windows, empty strings are undefined for environment variables, so that shouldn't be an issue. Linux does differentiate between unset environment variables and empty ones, but ultimately, it shouldn't matter. This is for debugging/testing only and isn't published anywhere. Anyone who relies on this _and_ misuses it deserves to have a broken build.

(That's also why we don't verify that it's really an integer before we try to coerce it into one.)